packages/expo-notifications/src/NotificationPermissionsModule.ts

```diff
@@ -1,6 +1,73 @@
+import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
+import { PermissionStatus } from 'unimodules-permissions-interface';
+
+import {
+  NativeNotificationPermissionsRequest,
+  NotificationPermissionsStatus,
+} from './NotificationPermissions.types';
 import { NotificationPermissionsModule } from './NotificationPermissionsModule.types';
 
+function convertPermissionStatus(
+  status?: NotificationPermission | 'prompt'
+): NotificationPermissionsStatus {
+  switch (status) {
+    case 'granted':
+      return {
+        status: PermissionStatus.GRANTED,
+        expires: 'never',
+        canAskAgain: false,
+        granted: true,
+      };
+    case 'denied':
+      return {
+        status: PermissionStatus.DENIED,
+        expires: 'never',
+        canAskAgain: false,
+        granted: false,
+      };
+    default:
+      return {
+        status: PermissionStatus.UNDETERMINED,
+        expires: 'never',
+        canAskAgain: true,
+        granted: false,
+      };
+  }
+}
+
+async function resolvePermissionAsync(shouldAsk: boolean): Promise<NotificationPermissionsStatus> {
+  if (!canUseDOM) {
+    return convertPermissionStatus('denied');
+  }
+
+  const { Notification = {} } = window as any;
+  if (typeof Notification.requestPermission !== 'undefined') {
+    let status = Notification.permission;
+    if (shouldAsk) {
+      status = await Notification.requestPermission();
+    }
+    return convertPermissionStatus(status);
+  } else if (
+    typeof navigator === 'undefined' ||
+    typeof navigator.permissions === 'undefined' ||
+    typeof navigator.permissions.query === 'undefined'
+  ) {
+    const query = await navigator.permissions.query({ name: 'notifications' });
```

Comment:
Can we use `navigator.permissions.query` if (`navigator` is `undefined` or `.permissions` is undefined or…)? (Isn't the if clause inverted?)